subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -401,25 +438,48 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx)
 	if (!aux) {
 		aux = aux_acquire();
 		if (!aux) {
+			if (sync_lll) {
+				struct ll_sync_set *sync;
+
+				sync = HDR_LLL2ULL(sync_lll);
+				sync->data_len += data_len;
+				ftr->aux_data_len = sync->data_len;
+				sync->data_len = 0U;
+			}
+
 			goto ull_scan_aux_rx_flush;
 		}
 
 		aux->rx_head = aux->rx_last = NULL;
+		aux->data_len = data_len;
 		lll_aux = &aux->lll;
 		lll_aux->is_chain_sched = 0U;
 
 		ull_hdr_init(&aux->ull);
 		lll_hdr_init(lll_aux, aux);
 
 		aux->parent = lll ? (void *)lll : (void *)sync_lll;
+	} else {
+		aux->data_len += data_len;
 	}
 
+	lll_aux->chan = aux_ptr->chan_idx;
+	lll_aux->phy = BIT(aux_ptr->phy);
```

Comment:
Is it related with data length fix?

Reply:
Tried keeping it following the mem_acquire code that is above. No other reasons, but reverted back to where it was before.